src/ring_theory/laurent_series.lean

```diff
@@ -200,23 +160,23 @@ section laurent_series
 
 variables [comm_semiring R] (p q : polynomial R)
 
-open polynomial laurent_series hahn_series
+open laurent_series hahn_series
 
 lemma coe_laurent : (p : laurent_series R) = of_power_series ℤ R p := rfl
 
 @[norm_cast] lemma coe_coe : ((p : power_series R) : laurent_series R) = p := rfl
 
 @[simp] lemma coe_laurent_zero : ((0 : polynomial R) : laurent_series R) = 0 :=
-by rw [coe_laurent, coe_zero, _root_.map_zero]
+by rw [coe_laurent, ←coe_to_power_series.ring_hom_apply, ←ring_hom.comp_apply, _root_.map_zero]
```

Comment:
Does this golf to
```suggestion
map_zero coe_to_power_series.ring_hom
```
Or similar?

Reply:
I pushed the golf

Reply:
Thanks. Is it possible to use some alg_hom property to likewise golf the smul lemma?

Reply:
Yes, if you know where the alg_homs are 

I'm a bit confused about this entire collection of lemmas though - why are we proving things about a transitive `coe` rather than letting `_root_.coe_coe` unpack it to a pair of coercions, and writing lemmas about those separately?

Reply:
There are alg_homs for both of these coercions. I can't remember the name of the smul "commuting" lemma however, what is it?

You make a good point. I had thought the simp-normal form for a double coercion was a single coercion. I was mistaken, it is the other way around. I can clean this up.